src/vs/editor/browser/services/openerService.ts

```diff
@@ -4,49 +4,61 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as dom from 'vs/base/browser/dom';
+import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
+import { LinkedList } from 'vs/base/common/linkedList';
 import { parse } from 'vs/base/common/marshalling';
 import { Schemas } from 'vs/base/common/network';
 import * as resources from 'vs/base/common/resources';
+import { equalsIgnoreCase } from 'vs/base/common/strings';
 import { URI } from 'vs/base/common/uri';
 import { ICodeEditorService } from 'vs/editor/browser/services/codeEditorService';
 import { CommandsRegistry, ICommandService } from 'vs/platform/commands/common/commands';
-import { IOpenerService, IOpener } from 'vs/platform/opener/common/opener';
-import { equalsIgnoreCase } from 'vs/base/common/strings';
-import { IDisposable } from 'vs/base/common/lifecycle';
-import { LinkedList } from 'vs/base/common/linkedList';
-import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
-import { localize } from 'vs/nls';
-import { IProductService } from 'vs/platform/product/common/product';
-import { IStorageService, StorageScope } from 'vs/platform/storage/common/storage';
-import Severity from 'vs/base/common/severity';
 import { ServiceIdentifier } from 'vs/platform/instantiation/common/instantiation';
+import { IOpener, IOpenerService, IValidator } from 'vs/platform/opener/common/opener';
 
-export class OpenerService implements IOpenerService {
+export class OpenerService extends Disposable implements IOpenerService {
 
 	_serviceBrand!: ServiceIdentifier<any>;
 
 	private readonly _opener = new LinkedList<IOpener>();
+	private readonly _validatorMap: { [k: string]: LinkedList<IValidator> } = {};
 
 	constructor(
 		@ICodeEditorService private readonly _editorService: ICodeEditorService,
 		@ICommandService private readonly _commandService: ICommandService,
-		@IStorageService private readonly _storageService: IStorageService,
-		@IDialogService private readonly _dialogService: IDialogService,
-		@IProductService private readonly _productService: IProductService
 	) {
-		//
+		super();
 	}
 
 	registerOpener(opener: IOpener): IDisposable {
 		const remove = this._opener.push(opener);
 		return { dispose: remove };
 	}
 
+	registerValidator(uriScheme: string, validator: IValidator): IDisposable {
```

Comment:
Why not just for all URLs, why scoped to scheme? We do not do that for the opener right?